utils/host/src/block_range.rs

```diff
@@ -42,3 +42,17 @@ pub async fn get_validated_block_range(
 
     Ok((l2_start_block, l2_end_block))
 }
+
+// Get a fixed recent (less than 2 weeks) block range.
+pub async fn get_rolling_block_range(
+    data_fetcher: &OPSuccinctDataFetcher,
+    range: u64,
+) -> Result<(u64, u64)> {
+    let header = data_fetcher.get_l2_header(BlockId::finalized()).await?;
+    let l2_block_time = data_fetcher.get_l2_block_time().await?;
+    let block_count_in_two_weeks = 14 * 24 * 60 * 60 / l2_block_time;
+    let l2_end_block = (header.number / block_count_in_two_weeks) * block_count_in_two_weeks;
```

Comment:
Why not just use `header.number` here? You're diving by, then multiply by `block_count_in_two_weeks`.